java-frontend/src/main/java/org/sonar/java/bytecode/cfg/BytecodeCFGBuilder.java

```diff
@@ -0,0 +1,234 @@
+      for (Instruction instruction : instructions) {
+        sb.append(index).append(": ").append(Printer.OPCODES[instruction.opcode]).append("\n");
+        index++;
+      }
+      sb.append("Jumps to: ");
+      successors.forEach(s -> sb.append("B").append(s.id).append(" "));
+      sb.append("\n");
+      return sb.toString();
+    }
+  }
+
+  private static class BytecodeCFGMethodVisitor extends MethodVisitor {
+    Map<Label, Block> blockByLabel = new HashMap<>();
+    private Block currentBlock;
+    private CFGBytecode cfg;
+
+    BytecodeCFGMethodVisitor(Symbol.MethodSymbol methodSymbol) {
+      super(Opcodes.ASM5);
+      cfg = new CFGBytecode(methodSymbol);
+      currentBlock = new Block(cfg);
+    }
+
+    @Override
+    public void visitInsn(int opcode) {
+      currentBlock.addInsn(opcode);
+    }
+
+    @Override
+    public void visitIntInsn(int opcode, int operand) {
+      currentBlock.addInsn(opcode);
+    }
+
+    @Override
+    public void visitVarInsn(int opcode, int var) {
+      currentBlock.addInsn(opcode);
+    }
+
+    @Override
+    public void visitFieldInsn(int opcode, String owner, String name, String desc) {
+      currentBlock.addInsn(opcode);
+    }
+
+    @Override
+    public void visitMethodInsn(int opcode, String owner, String name, String desc, boolean itf) {
+      currentBlock.addInsn(opcode);
+    }
+
+    @Override
+    public void visitInvokeDynamicInsn(String name, String desc, Handle bsm, Object... bsmArgs) {
+      currentBlock.addInsn(Opcodes.INVOKEDYNAMIC);
+    }
+
+    @Override
+    public void visitLdcInsn(Object cst) {
+      // FIXME (npe) LDC is generic information about constant but the real value of constant could be loaded into instructions
+      currentBlock.addInsn(Opcodes.LDC);
+    }
+
+    @Override
+    public void visitIincInsn(int var, int increment) {
+      currentBlock.addInsn(Opcodes.IINC);
+    }
+
+    @Override
+    public void visitMultiANewArrayInsn(String desc, int dims) {
+      currentBlock.addInsn(Opcodes.MULTIANEWARRAY);
+    }
+
+    @Override
+    public void visitTableSwitchInsn(int min, int max, Label dflt, Label... labels) {
+      blockByLabel.computeIfAbsent(dflt, l -> currentBlock.createSuccessor());
+      for (Label label : labels) {
+        blockByLabel.computeIfAbsent(label, l -> currentBlock.createSuccessor());
+      }
+    }
+
+    @Override
+    public void visitLookupSwitchInsn(Label dflt, int[] keys, Label[] labels) {
+      for (Label label : labels) {
+        blockByLabel.computeIfAbsent(label, l -> currentBlock.createSuccessor());
+      }
+    }
+
+    @Override
+    public void visitJumpInsn(int opcode, Label label) {
+      blockByLabel.computeIfAbsent(label, l -> currentBlock.createSuccessor());
+      currentBlock = currentBlock.createSuccessor();
+    }
+
+    @Override
+    public void visitLabel(Label label) {
+      currentBlock = blockByLabel.computeIfAbsent(label, l -> currentBlock.createSuccessor());
+    }
+
```

Comment:
`org.objectweb.asm.MethodVisitor#visitTypeInsn` should also be overriden to include NEW, ANEWARRAY, CHECKCAST, INSTANCEOF opcodes